src/commonMain/kotlin/exchange.dydx.abacus/output/input/AdjustIsolatedMarginInput.kt

```diff
@@ -0,0 +1,175 @@
+package exchange.dydx.abacus.output.input
+
+import exchange.dydx.abacus.protocols.ParserProtocol
+import exchange.dydx.abacus.utils.Logger
+import kollections.JsExport
+import kotlinx.serialization.Serializable
+
+@JsExport
+@Serializable
+data class AdjustIsolatedMarginInputOptions(
+    val needsSize: Boolean?,
+) {
+    companion object {
+        internal fun create(
+            existing: AdjustIsolatedMarginInputOptions?,
+            parser: ParserProtocol,
+            data: Map<*, *>?
+        ): AdjustIsolatedMarginInputOptions? {
+            Logger.d { "creating Adjust Isolated Margin Input Options\n" }
+
+            data?.let {
+                val needsSize = parser.asBool(data["needsSize"])
+
+                return if (existing?.needsSize != needsSize
+                ) {
+                    AdjustIsolatedMarginInputOptions(needsSize)
+                } else {
+                    existing
+                }
+            }
+            Logger.d { "Adjust Isolated Margin Input Options not valid" }
+            return null
+        }
+    }
+}
+
+@JsExport
+@Serializable
+data class AdjustIsolatedMarginInputSummary(
+    val crossFreeCollateral: Double?,
+    val crossMarginUsage: Double?,
+    val positionMargin: Double?,
+    val positionLeverage: Double?,
+    val liquidationPrice: Double?,
```

Comment:
are these values all actually nullable?

Reply:
They should be `nullable`. The summary will try to populate after each change to the input. If `amount` has yet to be filled out then is possible that some of these values are `null`.